SimTKcommon/Simulation/include/SimTKcommon/internal/StateImpl.h

```diff
@@ -35,6 +35,100 @@ contain any user-visible objects. **/
 
 namespace SimTK {
 
+//==============================================================================
+//                             DEPENDENT LIST
+//==============================================================================
+/* This class contains a set of downstream cache entries that are dependent 
+on the value of some prerequisite that is the owner of this object. 
+Prerequisites may be 
+  - continuous state variables q,u,z
+  - discrete variables, or
+  - upstream cache entries. 
+
+When the prerequisite's value changes or is explicitly invalidated, the
+notePrerequisiteChange() method is invoked, causing the invalidate() method of 
+each of the dependent cache entries to be called.
+
+CAUTION: upstream cache entries may be invalidated implicitly by their
+depends-on stage becoming invalid. For this to work, the depends-on stage
+of a downstream cache entry must be the same as the highest depends-on stage of 
+any of it upstream cache prerequisites. That way it is guaranteed to be 
+implicitly invalidated whenever any of its upstream prerequisites are.
+
+The dependents are expected to know their prerequisites so that they can
+remove themselves from any DependentLists they are on when deleted; and so 
+that they can be registered on the appropriate DependentLists when copied.
+*/
+class DependentList {
```

Comment:
In your PR description you suggested this class was named `DependentsList`. `DependentList` is slightly confusing since "dependent" can be either a noun or an adjective (while "dependents" is a noun). What about "PrereqList"?